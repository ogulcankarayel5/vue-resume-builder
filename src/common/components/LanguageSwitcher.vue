<template>
  <div class="fixed right-3 top-2">
    <select v-model="$i18n.locale" class="bg-gray-300">
      <option :value="locale.code" v-for="locale in locales" :key="locale.code">
        {{ locale.name }}
      </option>
    </select>
  </div>
</template>
<script>
import Vue from "vue";
import { getSupportedLocales } from "@/plugins/i18n/utils";

export default Vue.extend({
  name: "LanguageSwitcher",
  data: () => ({ locales: getSupportedLocales() }),
});
</script>

<style></style>
