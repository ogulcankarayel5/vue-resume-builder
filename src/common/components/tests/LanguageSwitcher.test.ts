import LanguageSwitcher from "@/common/components/LanguageSwitcher.vue";
import { shallowMount } from "@vue/test-utils";
import Vue from "vue";
import { createI18n } from 'vue-i18n'
const i18n =  createI18n({ locale: "en" });
const locales = [
  { code: "en", name: "English" },
  { code: "tr", name: "Turkish" },
];
const factory = (values = {}, other = {}) => {
  return shallowMount(LanguageSwitcher, {
    data() {
      return {
        locales,
        ...values,
      };
    },
    ...other,
  });
};
describe("LanguageSwitcher", () => {
  it("snapshot", () => {
    const wrapper = factory();
    expect(wrapper.element).toMatchSnapshot();
  });
  it("should render the correct number of languages", () => {
    const wrapper = factory();
    expect(wrapper.findAll('[data-test-id="locale"]').length).toBe(
      locales.length
    );
  });
  it("should change locale when select another option", () => {
    const wrapper = factory();
    wrapper.find('[data-test-id="localeSelect"]').setValue("tr");
    expect((i18n as any).locale).toBe("tr");
  });
});
