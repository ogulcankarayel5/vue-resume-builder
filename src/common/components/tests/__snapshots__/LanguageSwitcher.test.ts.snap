// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSwitcher snapshot 1`] = `
<div
  class="fixed right-3 top-2"
>
  <select
    class="bg-gray-300"
    data-test-id="localeSelect"
  >
    <option
      data-test-id="locale"
      value="en"
    >
      
      English
    
    </option>
    <option
      data-test-id="locale"
      value="tr"
    >
      
      Turkish
    
    </option>
  </select>
</div>
`;
